pkg/codegen/internal/test/testdata/simple-methods-schema/dotnet/Foo.cs

```diff
@@ -0,0 +1,150 @@
+// *** WARNING: this file was generated by test. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Pulumi.Serialization;
+
+namespace Pulumi.Example
+{
+    [ExampleResourceType("example::Foo")]
+    public partial class Foo : Pulumi.ComponentResource
+    {
+        /// <summary>
+        /// Create a Foo resource with the given unique name, arguments, and options.
+        /// </summary>
+        ///
+        /// <param name="name">The unique name of the resource</param>
+        /// <param name="args">The arguments used to populate this resource's properties</param>
+        /// <param name="options">A bag of options that control this resource's behavior</param>
+        public Foo(string name, FooArgs? args = null, ComponentResourceOptions? options = null)
+            : base("example::Foo", name, args ?? new FooArgs(), MakeResourceOptions(options, ""), remote: true)
+        {
+        }
+
+        private static ComponentResourceOptions MakeResourceOptions(ComponentResourceOptions? options, Input<string>? id)
```

Comment:
This could go in SDK instead of codegen'd code or no?

Reply:
To reduce the size of the generated code? I don't see why not aside from having to require a version of the `Pulumi` package where it exists. Might be easier to move it to the `Utilities` class that gets generated, so then there's only one emitted rather than one for each resource class.

Reply:
Yes, pretty much reducing the size of generated code is what I have in mind here, but this might be negligible. Which package it belongs will have more to do with how stable the logic is and where it's best versioned (which probably is here, not SDK). "only one emitted rather than one for each resource class." that sounds good if there's a material reduction on SDK size, which again there probably isn't. Thanks.